ably-ios/ARTAuth.m

```diff
@@ -172,51 +190,77 @@ - (instancetype)initWithRest:(ARTRest *)rest options:(ARTAuthOptions *)options {
         _tokenCbs = nil;
         _tokenRequest = nil;
 
-        if (nil != options.keyValue && nil == options.clientId) {
-            _authMethod = ARTAuthMethodBasic;
+        
+        
+        if (nil != options.keyValue /*&& nil == options.clientId*/) {
+            [ARTLog debug:@"ARTAuth: setting up auth method Basic"];
             _basicCredentials = [NSString stringWithFormat:@"Basic %@", [[[NSString stringWithFormat:@"%@:%@", options.keyId, options.keyValue] dataUsingEncoding:NSUTF8StringEncoding] base64EncodedStringWithOptions:0]];
             _keyId = options.keyId;
             _keyValue = options.keyValue;
-        } else {
-            _authMethod = ARTAuthMethodToken;
+        }
+        else {
+            [ARTLog warn:@"ARTAuth Error: cannot set up basic auth without a value keyValue. "];
+        }
+    
+        if(options.useTokenAuth) {
```

Comment:
There are many conditions where token auth should be used even when `useTokenAuth` option has not been set to true.  See https://github.com/ably/ably-java/blob/master/src/io/ably/rest/Auth.java#L584-L620.  In most cases, token auth is used in fact.